Sources/XcodeExport/Model/XcodeExportExtensions.swift

```diff
@@ -0,0 +1,225 @@
+        let name = light.name
+        let dirURL = directory.appendingPathComponent("\(name).imageset")
+
+        let lightPack = light.packForXcode()
+        let darkPack = dark?.packForXcode()
+
+        let properties = { () -> XcodeAssetContents.TemplateProperties? in
+            if let preservesVector = preservesVector {
+                return XcodeAssetContents.TemplateProperties(
+                    preservesVectorRepresentation: preservesVector ? true : nil
+                )
+            } else {
+                return nil
+            }
+        }()
+
+        let lightFiles = lightPack?.makeImageFileContents(to: dirURL, appearance: .light) ?? []
+        let darkFiles = darkPack?.makeImageFileContents(to: dirURL, appearance: .dark) ?? []
+
+        let lightAssetContents = lightPack?.makeXcodeAssetContentsImageData(appearance: .light) ?? []
+        let darkAssetContents = darkPack?.makeXcodeAssetContentsImageData(appearance: .dark) ?? []
+
+        let contentsFileContents = try XcodeAssetContents(
+            images: lightAssetContents + darkAssetContents,
+            properties: properties
+        ).makeFileContents(to: dirURL)
+
+        return [contentsFileContents] + lightFiles + darkFiles
+    }
+
+}
+
+// MARK: Image
+
+extension Image {
+
+    func makeFileContents(to directory: URL, appearance: Appearance? = nil) -> FileContents {
+        let fileURL = makeFileURL(scale: scale, appearance: appearance)
+        let destination = Destination(directory: directory, file: fileURL)
+        return FileContents(destination: destination, sourceURL: url)
+    }
+
+    func makeXcodeAssetContentsImageData(scale: Scale, appearance: Appearance? = nil) -> XcodeAssetContents.ImageData {
+        let filename = makeFileURL(scale: scale, appearance: appearance).absoluteString
+        let xcodeIdiom = idiom.flatMap { XcodeAssetIdiom(rawValue: $0) } ?? .universal
+        let appearances = appearance.flatMap { $0 == .dark ? [XcodeAssetContents.DarkAppeareance()] : nil }
+        let scaleString = scale.string
+
+        return XcodeAssetContents.ImageData(
+            idiom: xcodeIdiom,
+            scale: scaleString,
+            appearances: appearances,
+            filename: filename
+        )
+    }
+
+    private func makeFileURL(scale: Scale, appearance: Appearance? = nil) -> URL {
+        var urlString = name
+
+        if let idiom = idiom, !idiom.isEmpty {
+            urlString.append("~\(idiom)")
+        }
+
+        switch appearance {
+        case .light:
+            urlString.append("L")
+        case .dark:
+            urlString.append("D")
+        default:
+            break
+        }
+
+        if let scaleString = scale.string {
+            urlString.append("@\(scaleString)")
+        }
+
+        return URL(string: urlString)!.appendingPathExtension(format)
+    }
+
+    fileprivate func isValidForXcode(scale: Scale) -> Bool {
```

Comment:
Why did you choose `fileprivate` instead of `private`?

Reply:
The function is called in other extension, so i need to make it `fileprivate`.